phoenix-core/src/main/java/org/apache/phoenix/compile/QueryCompiler.java

```diff
@@ -653,9 +653,12 @@ protected QueryPlan compileSingleFlatQuery(
         PTable table = tableRef.getTable();
 
         ParseNode viewWhere = null;
-        if (table.getViewStatement() != null) {
+        if (table.getViewTTL() > 0) {
```

Comment:
For clarity, can you please add a comment here where we say that viewTTL is expected to be non-zero only for views or view indexes?